In [95]:
import numpy as np
import pandas as pd

In [96]:
df = pd.read_csv('train.csv')

In [97]:
df.head()

col1        col2  output
0 -38.366669  113.000000       1
1 -39.818572  123.943369       1
2 -55.638240  114.551862       1
3 -36.378005   95.616515       1
4 -39.366669  112.000000       1

In [98]:
target = df['output']
data  = df.drop("output",axis = 1)

In [99]:

data.head()

col1        col2
0 -38.366669  113.000000
1 -39.818572  123.943369
2 -55.638240  114.551862
3 -36.378005   95.616515
4 -39.366669  112.000000

In [100]:
target.head()

0    1
1    1
2    1
3    1
4    1
Name: output, dtype: int64

# Standardization:

In [101]:
from sklearn.preprocessing import StandardScaler
standardized_data = StandardScaler().fit_transform(data)
print(standardized_data.shape)

(1278, 2)


# Training the model:

In [102]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(standardized_data,target,test_size = 0.20,random_state = 0)                                                   

# Logistic Regression:

In [131]:
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression()
classifier.fit(X_train, y_train)

LogisticRegression()

In [132]:
from pickle import dump,load

In [133]:
dump(classifier,open('logistic.pkl','wb'))

In [134]:
classifier = load(open('logistic.pkl','rb'))

# Prediction:

In [135]:
y_predicted = classifier.predict(X_test)

# Evaluation:

In [136]:
from sklearn import metrics
metrics.accuracy_score(y_test, y_pred)

0.61328125

# Support Vector Machine(rbf):

In [106]:
#X_test1= ['X_test']
#X_train1= ['X_train']
#y_test1 = ['y_test']
#y_train1= ['y_train']

In [125]:
# default (kernel  = 'rbf')
from sklearn.svm import SVC
clf = SVC(kernel='rbf',C =10)
clf.fit(X_train, y_train)

SVC(C=10)

In [126]:
from pickle import dump,load

In [127]:
dump(clf1,open('svmr.pkl','wb'))

In [128]:
clf1 = load(open('svmr.pkl','rb'))

In [129]:
y_pre = clf.predict(X_test)

In [109]:
from sklearn import metrics
metrics.accuracy_score(y_test, y_pre)

1.0

# Support Vector Machine(linear):

In [116]:
from sklearn.svm import SVC

In [117]:
classifier1 = SVC(kernel='linear',C=10)
classifier1.fit(X_train, y_train)

SVC(C=10, kernel='linear')

In [118]:
from pickle import dump,load

In [120]:
dump(classifier1,open('svml.pkl','wb'))

In [121]:
classifier1 = load(open('svml.pkl','rb'))

In [122]:
y_test_pred = classifier1.predict(X_test)

In [123]:
from sklearn import metrics
metrics.accuracy_score(y_test, y_test_pred)

0.61328125

# Observation:


### Support Vector Machine (rbf) is the best model compared to SVM(linear) and Logistic model because the accuracy score for SVM(rbf) is 100% where accuracy for SVM(linear) and Logistic model are 61%.